src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -451,11 +457,25 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                     continue;
                 }
 
-                if (string.Equals(arg, shared, StringComparison.OrdinalIgnoreCase))
+                if (arg.StartsWith(shared, StringComparison.OrdinalIgnoreCase))
                 {
+                    var hasValue = arg.Length > shared.Length && (arg[shared.Length] == ':' || arg[shared.Length] == '=');
```

Comment:
I still don't see how this is a better approach.   Consider the code sample you originally posted:

```
var values = arg.Split(new[] { ':', '='}, 2)
```

Now think about the input `/sharedextra=foo`   That would pass the condition you set but be invalid input.  

Note: i mostly mimic'd the existing approach for parsing `/keepalive` just above this code.